### PR TITLE
Fix Load and Save in TASPaletteEditor

### PR DESCRIPTION
**Pull request by linev**
When saving palette, interpreter with wrong arguments was invoked.
Call directly `gROOT->SaveObjectAs` with correct argument.

Use TString for file name operations

When reading palette - check return values and also use TString for filename operations

### CODE REVIEW

**Comment by dpiparo** [comment]
Thanks @linev . The original PR evolved (https://github.com/root-project/root/pull/14960) maybe we can wait it's merged and rebase this one on top of master?

**Comment by linev** [comment]
@dpiparo I miss your PR. Of course, I will wait until then.

**Review by couet — APPROVED** [review]

File: graf2d/asimage/src/TASPaletteEditor.cxx [hunk 1]
```diff
@@ -467,42 +463,41 @@ void TASPaletteEditor::Open()
...
 
-      TFile *fsave = new TFile(fn, "READ");
+      TFile *fsave = new TFile(fn.Data(), "READ");
       if (!fsave->IsOpen()) {
          delete fsave;
          return;
       }
 
-      newPalette = (TImagePalette*)fsave->Get("TImagePalette");
+      newPalette = (TImagePalette *)fsave->Get("TImagePalette");
```
Comment (vepadulano) [inline graf2d/asimage/src/TASPaletteEditor.cxx:497]: Is it necessary here to allocate on the heap here? Can't we just create a TFile on the stack?

Comment (linev) [reply graf2d/asimage/src/TASPaletteEditor.cxx:497]: One should use `TFile::Open` here. I will adjust commit.

**Pull request merged**